pkg/kubelet/kubelet_pods.go

```diff
@@ -1447,7 +1493,10 @@ func getPhase(pod *v1.Pod, info []v1.ContainerStatus, podIsTerminal bool) v1.Pod
 	}
 
 	switch {
-	case pendingInitialization > 0:
+	case pendingInitialization > 0 &&
+		// This is needed to handle the case where the pod has been initialized but
+		// the restartable init containers are restarting.
```

Comment:
nit slight-reword:

// This is needed to handle the case where the pod has been initialized but the restartable init containers are restarting and the pod should *not* be placed back into `v1.PodPending` since the regular init containers have run.